### PR TITLE
922_sort_array_by_parity_2

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code.

### CODE REVIEW

File: 922_sort_array_by_parity_2.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from typing import List
+
+
+class Solution:
+    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
```
Comment (ryujumoto) [inline 922_sort_array_by_parity_2.py:8]: This solution is inefficient. Solve this with more efficient solution.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I changed the way. Please check my code.

File: 922_sort_array_by_parity_2.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# time: O(n)
+# space: O(n)
+from typing import List
+
+
+class Solution:
+    
+    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
+        even, odd, full = 0, 1, len(nums) - 1
```
Comment (ryujumoto) [inline 922_sort_array_by_parity_2.py]: full -> nums_length

File: 922_sort_array_by_parity_2.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# time: O(n)
+# space: O(n)
```
Comment (ryujumoto) [inline 922_sort_array_by_parity_2.py]: space is O(1).

**Comment by Rwatana** [comment]
I revised.Please check my code.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**